core/stream/convert.go

```diff
@@ -90,6 +90,9 @@ func Convert(dst, src *Format, data []byte) ([]byte, error) {
 	// Calculate min/max if floats
 	min, max := float64(math.MaxFloat64), -float64(math.MaxFloat64)
 	for _, m := range mappings {
+		if m.src.component == nil {
+			continue
+		}
```

Comment:
Instead of doing the continue, move the error return from the below loop here (and adjust the comment). That makes it a proper early exit on error, so we won't ever convert any channels if we end up returning an error for a missing mapping anyway, so an overall improvement, too.